pkg/codec/x264/bridge.h

```diff
@@ -0,0 +1,88 @@
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <x264.h>
+
+typedef struct Slice {
+  unsigned char *data;
+  int data_len;
+} Slice;
+
+typedef struct Encoder {
+  x264_t *h;
+  x264_picture_t pic_in, pic_out;
+  x264_param_t param;
+} Encoder;
+
+Encoder *enc_new(x264_param_t param) {
+  Encoder *e = (Encoder *)malloc(sizeof(Encoder));
+
+  if (x264_param_default_preset(&e->param, "veryfast", "zerolatency") < 0)
+    goto fail;
+
+  /* Configure non-default params */
+  e->param.i_csp = param.i_csp;
+  e->param.i_width = param.i_width;
+  e->param.i_height = param.i_height;
+  e->param.i_fps_num = param.i_fps_num;
+  e->param.i_fps_den = 1;
+  // Intra refres:
+  e->param.i_keyint_max = param.i_fps_num;
+  e->param.b_intra_refresh = 1;
+  // Rate control:
+  e->param.rc.i_rc_method = X264_RC_CRF;
+  e->param.rc.f_rf_constant = 25;
+  e->param.rc.f_rf_constant_max = 35;
+  // For streaming:
+  e->param.b_repeat_headers = 1;
+  e->param.b_annexb = 1;
+
+  if (x264_param_apply_profile(&e->param, "baseline") < 0)
+    goto fail;
+
+  if (x264_picture_alloc(&e->pic_in, param.i_csp, param.i_width, param.i_height) < 0)
+    goto fail;
+
+#undef fail
+#define fail fail2
```

Comment:
`fail` label is not used later. Just `undef` seems enough to avoid leaking.

Reply:
I've removed it, instead uses fail2 directly instead.